SparkPost/2web2ui
# FE-296 Catch component errors with React Error Boundary

How to test:
- Full page: throw an exception from public or protected router.
- In Page: throw an exception from any pages in `src/pages` or any component referenced in there. 
- In development, you'll still see the error overlay. Seems there isn't a way to hide it (it was making me crazy thinking my error boundary wasn't working). Just close it to see the page rendered by boundary component. For same reason, I believe, the errors will be logged to sentry even if we remove `ErrorTracker.report({ info }, error);` from `ErrorBoundary.js`.



Notes:
- Two error boundaries. One for whole app, another for just pages. So in that case the nav will be still rendered. If the error is specific to a page, users can still visit other pages. App boundary will, however, show full page message
- Full page one probably needs some styling. Let me know about the copy of exact message. 
- For full page error, i've put a button to reload. May be here (and in page error as well), we can [ask for user feedback](https://docs.sentry.io/learn/user-feedback/) :mindblown:
- Doesn't look like `report` in errorTracker working as expected (or i need to understand it). Seems the extra contexts aren't visible in sentry. may be we should expose apis to set context. that way we can make it explicit that error is coming from boundaries. 

Other note:
Seems App.js, index.js isn't under test coverage. Any reason not to?
